### PR TITLE
(RK-195) Serialize resolved env-map to disk after deploy

### PR DESCRIPTION
**Pull request by scotje**
Also converts the .blob_at functionality to be portable across providers.

### CODE REVIEW

File: lib/r10k/git/rjgit.rb [hunk 1]
```diff
@@ -60,6 +60,13 @@ def rev_parse(ver, opts = {})
         return commit.id
       end
 
+      def blob_at(ver, path, opts = {})
+        repo = ::RJGit::Repo.new(opts[:git_dir], is_bare: true)
```
Comment (adreyer) [inline lib/r10k/git/rjgit.rb:103]: should we check for this opt what happens otherwise?

**Comment by adreyer** [comment]
I think blob_at should probably require a git_dir since it's unlikely to behave well without one.

**Comment by scotje** [comment]
It seems like #blob_at doesn't behave any worse than anything else when it doesn't get a `:git_dir` so I'm inclined to leave it be for now and address the option validation more comprehensively in a separate ticket.

File: lib/r10k/git/rjgit.rb [hunk 2]
```diff
@@ -60,6 +60,13 @@ def rev_parse(ver, opts = {})
...
+        blob = repo.blob(path, ver)
+
+        return blob.data
```
Comment (andersonmills) [inline lib/r10k/git/rjgit.rb]: Do we want to raise an Error on failure as we're doing in the shellgit version of this?

Comment (scotje) [inline lib/r10k/git/rjgit.rb]: Yeah, let me dig down and figure out what hideously long exception classes that can raise....

File: lib/r10k/git/shellgit.rb [hunk 3]
```diff
@@ -67,6 +67,20 @@ def rev_parse(rev, opts = {})
...
 
+      def blob_at(treeish, path, opts = {})
+        opts = { raise_on_fail: false }.merge(opts)
+
+        cmd = ["show", "#{treeish}:#{path}"]
+
+        result = git(cmd, opts)
+
+        if result.success?
+          return result.stdout.strip
```
Comment (andersonmills) [inline lib/r10k/git/shellgit.rb]: Should the shellgit and rjgit versions of this return the same thing?

Comment (andersonmills) [inline lib/r10k/git/shellgit.rb]: Okay, on further inspection, these only differ by an ending newline. stdout gets chomped in subprocess.

**Comment by andersonmills** [comment]
:+1: I believe!

I believe I can fly!

**Pull request merged**